scene/gui/dialogs.cpp

```diff
@@ -29,11 +29,12 @@
 /**************************************************************************/
 
 #include "dialogs.h"
-#include "dialogs.compat.inc"
 
 #include "scene/gui/line_edit.h"
 #include "scene/theme/theme_db.h"
 
+#include "dialogs.compat.inc"
```

Comment:
This was moved because otherwise, `LineEdit` is a forward declaration.
It probably should have been here from the start :)

Reply:
That's something that might warrant a codebase wide update then and a change to the documentation: https://docs.godotengine.org/en/latest/contributing/development/handling_compatibility_breakages.html#id2

Because this is a policy currently that the `<class>.compat.inc` should be next to `<class>.h`.

If that's not a good idea, and it would be better to always have it come last, then this should be made consistent.

But the other option here could just be to include `line_edit.h` in `dialogs.compat.inc` where it's needed?

Reply:
Oh, good to know!
Then I think i'll do what you suggested and include `line_edit.h` in `dialogs.compat.inc`.

I do think `.compat.inc` might be better placed behind the includes, since it's not meant to be used as a standalone header. But I'll probably suggest that in a GIP separately.